File: lab1/send_recv.c
#include <stdio.h>
#include "mpi.h"

int main(int argc, char *argv[] )
{
  int rank, value, size;
  MPI_Status status;

  //Add call to mpi init
  //Add call to get rank
  //Add call to get size of communicator

  //Replace the commented lines with MPI calls
  if (rank == 0) {
    value = 5;
    printf( "Process %d sending %d\n", rank, value );
    //Insert MPI command to send value to the next rank
  } else {
    //Insert MPI command to receive value from the previous rank
    printf( "Process %d got %d\n", rank, value );
    if (rank < size - 1)
      //Insert MPI command to send value to the next rank
  }

  MPI_Finalize( );
  return 0;
}

File: lab1/send_recv.f90
program send_recv

implicit none

include "mpif.h"

integer :: rank, value, comm_size, ierr
integer :: istatus(MPI_STATUS_SIZE)

!Add call to mpi init
!Add call to get rank
!Add call to get size of communicator

value = 0
!Replace the commented lines with MPI calls
if (rank == 0) then
   value = 5 ! we set this value directly for rank 0 process
   !Insert MPI command to send value to the next rank
   print *, "Process ", rank, " sent ", value
else
   !Insert MPI command to receive value from the previous rank
   print *, "Process ", rank, " got ", value
   if (rank < comm_size-1) then
      !Insert MPI command to send value to the next rank
      print *, "Process ", rank, " sent ", value
   end if
end if

call MPI_Barrier(MPI_COMM_WORLD, ierr)
call MPI_Finalize(ierr)

end program
